packages/frontend/src/components/wallet/ActivitiesWrapper.js

```diff
@@ -84,28 +86,36 @@ const StyledContainer = styled.div`
     }
 `;
 
-const Activities = ({ transactions, accountId, getTransactionStatus }) => {
+const ActivitiesWrapper = () => {
+    const dispatch = useDispatch();
+
     const [transactionHash, setTransactionHash] = useState();
     const activityLoader = actionsPending(['GET_TRANSACTIONS', 'REFRESH_ACCOUNT_OWNER']);
-    const transaction = useSelector((state) => selectOneTransactionByHash(state, { accountId, hash: transactionHash }));
+
+    const accountId = useSelector(state => selectAccountId(state));
+    const transactions = useSelector(state => selectTransactionsByAccountId(state, { accountId }));
+    const transaction = useSelector(state => selectOneTransactionByHash(state, { accountId, hash: transactionHash }));
+
+    useEffect(() => {
+        if (accountId) {
+            dispatch(getTransactions(accountId));
+        }
+    }, [accountId]);
 
     return (
         <StyledContainer>
             <h2 className={classNames({'dots': activityLoader})}><Translate id='dashboard.activity' /></h2>
-            {transactions
-                ? transactions.map((transaction, i) => (
-                    <ActivityBox
-                        key={i}
-                        transaction={transaction}
-                        actionArgs={transaction.args}
-                        actionKind={transaction.kind}
-                        receiverId={transaction.receiver_id}
-                        accountId={accountId}
-                        setTransactionHash={setTransactionHash}
-                    />
-                ))
-                : null
-            }
+            {transactions.map((transaction, i) => (
+                <ActivityBox
+                    key={transaction.hash}
```

Comment:
Since we use `${hash}-${kind}` as a unique key, this should use the same thing.
https://github.com/near/near-wallet/blob/3e356525ee4b5a9bd054fc4aae3a354916a331eb/packages/frontend/src/components/wallet/ActivityBox.js#L126

Reply:
Done in #2031 :
https://github.com/near/near-wallet/pull/2031/commits/cb252e8c19a83869dbaffe6b8ac5c4e2c81f4390